application-xpoll-api/src/main/java/com/xwiki/xpoll/internal/rest/DefaultXPollResource.java

```diff
@@ -67,7 +67,7 @@
     @Override
     public Response saveXPollAnswers(String wikiName, String spaces, String pageName) throws XWikiRestException
     {
-        if (!contextualAuthorizationManager.hasAccess(Right.EDIT)) {
+        if (!contextualAuthorizationManager.hasAccess(Right.EDIT, getXWikiContext().getUserReference())) {
```

Comment:
It would say that the user didn't have any right. I debugged and tried for the VIEW and EDIT rights, but they returned false. Using this seemed to have solved the problem.

Reply:
> Using this seemed to have solved the problem.

This is not enough. You need to understand **why** it works (if it works). See https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-11.10/xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-api/src/main/java/org/xwiki/security/authorization/ContextualAuthorizationManager.java#L50 versus https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-11.10/xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-api/src/main/java/org/xwiki/security/authorization/ContextualAuthorizationManager.java#L73 . The second parameter is not the user reference but the entity on which you want to check the access right. So you're checking if the current user has edit right on their profile page (because the user reference is the reference of the user profile document). I doubt this is your intention.

The issue might be that the current document is not set on the context, so ``hasAccess(Right.EDIT)`` doesn't know which is the target document for which to check the access right. In this case (but please check first), and if the context user is set, then you should:
* either set the context document before this call (might be needed elsewhere later)
* or simply pass the poll reference as the second argument, because you want to check if the current user can edit the poll document

Reply:
The document wasn't set indeed, as `context.getDocument()` returned `null`. I simply passed the poll reference as the second argument, since the rest of the code didn't make use of the document set in the context.